x-pack/metricbeat/module/autoops_es/utils/utils.go

```diff
@@ -97,8 +96,13 @@ func PartitionByMaxValue[T any](limit int, items []T, valueExtractor func(T) int
 		itemKey := valueExtractor(item)
 		sortedValues[itemKey] = append(sortedValues[itemKey], item)
 	}
-	allKeys := slices.Collect(maps.Keys(sortedValues))
-	slices.Sort(allKeys)
+
+	allKeys := make([]int, 0, len(sortedValues))
+	for key := range sortedValues {
+		allKeys = append(allKeys, key)
+	}
+	sort.Ints(allKeys)
```

Comment:
`maps.Keys` gives us an iterator equivalent to the `allKeys` slice. Then, `slices.Sort(allKeys)` does the same sort and, in fact, [in Go 1.22+, `sort.Ints` calls `slices.Sort`](https://pkg.go.dev/sort#Ints).

```go
	allKeys := slices.Collect(maps.Keys(sortedValues))
	slices.Sort(allKeys)
```